src/NovelRT/ResourceManagement/Desktop/DesktopResourceLoader.cpp

```diff
@@ -269,20 +269,22 @@ namespace NovelRT::ResourceManagement::Desktop
         BinaryPackage package{};
         package.memberMetadata = {};
 
-        jsoncons::json j = jsoncons::bson::decode_bson<jsoncons::json>(buffer);
-        jsoncons::json metadata = j["memberMetadata"];
+        auto j = nlohmann::json::from_bson(buffer);
+        auto metadata = j["memberMetadata"];
 
-        for (auto&& obj : metadata.array_value())
+        for (auto&& obj : metadata.array())
         {
-            BinaryMemberMetadata newMemberMetadata{
-                obj["name"].as<std::string>(), static_cast<BinaryDataType>(obj["type"].as<uint32_t>()),
-                obj["location"].as<size_t>(),  obj["sizeOfTypeInBytes"].as<size_t>(),
-                obj["length"].as<size_t>(),    obj["sizeOfSerialisedDataInBytes"].as<size_t>()};
+            BinaryMemberMetadata newMemberMetadata{obj.value<std::string>("name", std::string()),
+                                                   obj.value<BinaryDataType>("type", BinaryDataType::NullOrUnknown),
```

Comment:
What are the possibly `BinaryDataType`s here?

Reply:
If you mean how it's serialized: it's serialized as an integer by default,they can be serialized as strings as seen [here](https://json.nlohmann.me/features/enum_conversion/).

if you mean the values `BinaryDataType` can represent, then a full list can be seen [here](https://github.com/novelrt/NovelRT/blob/main/include/NovelRT/ResourceManagement/BinaryDataType.h). examples are:
* Boolean
* Int32
* Int64

Reply:
CC @RubyNova - should we be typing as UInt32 here? I see the static cast has been dropped, but I'm not sure if its necessary anymore (CI appears to pass but idk if we specifically test this)

Reply:
LGTM